core/src/test/java/org/infinispan/commands/module/TestGlobalConfigurationBuilder.java

```diff
@@ -47,7 +48,9 @@ public TestGlobalConfigurationBuilder testCacheComponent(String cacheName, Strin
    }
 
    public TestGlobalConfigurationBuilder addCacheStartingCallback(Consumer<ComponentRegistry> callback) {
-      this.attributes.attribute(TestGlobalConfiguration.CACHE_STARTING_CALLBACKS).get().add(callback);
+      List<Consumer<ComponentRegistry>> callbacks = this.attributes.attribute(TestGlobalConfiguration.CACHE_STARTING_CALLBACKS).get();
+      callbacks.add(callback);
+      this.attributes.attribute(TestGlobalConfiguration.CACHE_STARTING_CALLBACKS).set(callbacks);
```

Comment:
Why is this change necessary?

Reply:
It's because it marks the attribute as modified